File: frontend/src/types/bulk-upload.ts
export type BulkUploadStatus =
  | "PENDING"
  | "IN_PROGRESS"
  | "COMPLETED"
  | "PARTIALLY_FAILED";

export interface BulkUploadJob {
  id: string;
  vendor_id: string;
  total_reports: number;
  processed_count: number;
  failed_count: number;
  status: BulkUploadStatus;
  created_at: string;
  updated_at: string;
}

export interface BulkUploadReportStatus {
  report_id: string;
  status: string;
  property_address: string | null;
}

File: backend/app/schemas/bulk_upload.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class BulkUploadJobResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    vendor_id: UUID
    total_reports: int
    processed_count: int
    failed_count: int
    status: str
    created_at: datetime
    updated_at: datetime


class BulkUploadReportStatus(BaseModel):
    """Per-report status within a bulk job."""
    report_id: UUID
    status: str
    property_address: str | None = None
